cowprotocol/cowswap
# feat(hook store): Add CoW AMM withdraw iframe dapp

# Summary

- Adds the CoW AMM Withdraw Pre Hook into the hook store registry by integrating with Bleu repository (https://github.com/bleu/cow-hooks-dapps/tree/main/apps/withdraw-pool)

<img width="464" alt="image" src="https://github.com/user-attachments/assets/579b8b76-0d97-4827-824a-5d7577b8f3a3">
<img width="461" alt="image" src="https://github.com/user-attachments/assets/a9441564-15a9-425f-99c1-61b7fc7db4cb">
<img width="463" alt="image" src="https://github.com/user-attachments/assets/52d87857-a8e2-4237-bfe4-b7851b165a81">
<img width="464" alt="image" src="https://github.com/user-attachments/assets/07d6d9d7-e35b-446c-9e43-6c04f548d196">
<img width="459" alt="image" src="https://github.com/user-attachments/assets/03f62b5a-c2a1-4b56-a95d-95ab1e95e037">

# To Test

1. Go the hooks tab on Sepollia chain
2. Connect your wallet and switch to Sepollia chain
3. In Hooks > Add Pre-Hook Action, the withdraw hook should appear in the "All hooks" section. 
4. To fully test the featureyou must have liquidity on one CoW AMM pool. I already added some for these addresses:
  - 0x8FAb71C0d4272698A3B2d1F3Ed5FC3c1B9b3E531
  - 0xFF714b8b0e2700303eC912BD40496C3997ceEa2b
  - 0xFF714b8b0e2700303eC912BD40496C3997ceEa2b

# Note:

- Smart Wallets are not supported yet since we have to define how the transaction flow would work (on the approval). If you have some idea, please share with us 😃
